superset/assets/src/dashboard/components/PropertiesModal.jsx

```diff
@@ -0,0 +1,282 @@
+      }));
+      this.onOwnersChange(initialSelectedValues);
+    });
+  }
+
+  onOwnersChange(value) {
+    this.setState(state => ({
+      values: {
+        ...state.values,
+        owners: value,
+      },
+    }));
+  }
+
+  onChange(e) {
+    const { name, value } = e.target;
+    this.setState(state => ({
+      values: {
+        ...state.values,
+        [name]: value,
+      },
+    }));
+  }
+
+  toggleAdvanced() {
+    this.setState(state => ({
+      isAdvancedOpen: !state.isAdvancedOpen,
+    }));
+  }
+
+  save(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    const owners = this.state.values.owners.map(o => o.value);
+    SupersetClient.put({
+      endpoint: `/api/v1/dashboard/${this.props.dashboardInfo.id}`,
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        ...this.state.values,
+        owners,
+      }),
+    })
+      .then(({ json }) => {
+        this.props.addSuccessToast(t('The dashboard has been saved'));
+        this.props.onDashboardSave({
+          title: json.result.dashboard_title,
+          slug: json.result.slug,
+          jsonMetadata: json.result.json_metadata,
+          ownerIds: json.result.owners,
+        });
+        this.props.onHide();
+      })
+      .catch(response =>
+        getClientErrorObject(response).then(({ error, statusText }) => {
+          this.dialog.show({
+            title: 'Error',
+            bsSize: 'medium',
+            bsStyle: 'danger',
+            actions: [Dialog.DefaultAction('Ok', () => {}, 'btn-danger')],
+            body: error || statusText || t('An error has occurred'),
+          });
+        }),
+      );
+  }
+
+  render() {
+    const { userOptions, values, isOwnersLoaded, isAdvancedOpen } = this.state;
+    return (
+      <Modal show={this.props.show} onHide={this.props.onHide} bsSize="lg">
+        <form onSubmit={this.save}>
+          <Modal.Header closeButton>
+            <Modal.Title>
+              <div>
+                <span className="float-left">{t('Dashboard Properties')}</span>
+              </div>
+            </Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Row>
+              <Col md={12}>
+                <h3>Basic Information</h3>
+              </Col>
+            </Row>
+            <Row>
+              <Col md={6}>
+                <label className="control-label" htmlFor="embed-height">
+                  {t('Title')}
+                </label>
+                <FormControl
+                  name="dashboard_title"
+                  type="text"
+                  bsSize="sm"
+                  value={values.dashboard_title}
+                  onChange={this.onChange}
+                />
+              </Col>
+              <Col md={6}>
+                <label className="control-label" htmlFor="embed-height">
+                  {t('URL Slug')}
+                </label>
+                <FormControl
+                  name="slug"
```

Comment:
Can we include the description on this case is `To get a readable URL for your dashboard`